pkg/v1/config/context.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package config
+
+import (
+	"fmt"
+
+	configv1alpha1 "github.com/vmware-tanzu/tanzu-framework/apis/config/v1alpha1"
+)
+
+// GetContext by name.
+func GetContext(name string) (*configv1alpha1.Context, error) {
+	cfg, err := GetClientConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	return cfg.GetContext(name)
+}
+
+// ContextExists tells whether the Context by the given name exists.
+func ContextExists(name string) (bool, error) {
+	cfg, err := GetClientConfig()
+	if err != nil {
+		return false, err
+	}
+
+	return cfg.HasContext(name), nil
+}
+
+// AddContext adds a Context to the config.
+func AddContext(c *configv1alpha1.Context, setCurrent bool) error {
+	cfg, err := GetClientConfig()
```

Comment:
`AcquireTanzuConfigLock()` needs to be added before changing the config. Similarly in case of Remove method as well!

Reply:
Also in `SetCurrentContext`

Reply:
Sure, will add it! Is there a reason why the acquire lock is not part of `StoreClientConfig `? That way not all callers do not have to worry about the lock

Reply:
@sathyanarays Added the locks!